src/Logic.hs

```diff
@@ -1186,51 +1246,36 @@ pushCandidate (pullRequestId, pullRequest) newHead@(Sha sha) state =
       commentToUser = leaveComment pullRequestId . (<>) ("@" <> approvedBy <> " ")
   in assert (isJust $ Pr.approval pullRequest) $ do
     let approvalKind = Pr.approvedFor approval
-    pushResult <- if Pr.needsTag approvalKind
-      then do
-        versionOrBadTag <- getLatestVersion newHead
-        config <- getProjectConfig
-        case versionOrBadTag of
-          Left (TagName bad) -> do
-            commentToUser ("Sorry, I could not tag your PR. The previous tag `" <> bad <> "` seems invalid")
-            tryPromote prBranch newHead
-          Right v -> do
-            let previousTag = versionToTag v
-            mChangelog <- getChangelog previousTag newHead
-            let
-              tagName = versionToTag $ v + 1
-              changelog = fromMaybe "Failed to get the changelog" mChangelog
-              tagMessage = messageForTag tagName approvalKind changelog
-            (tagResult, pushResult) <- tryPromoteWithTag prBranch newHead tagName tagMessage
-            when (pushResult == PushOk) $ commentToUser $
-              case tagResult of
-                Left err -> "Sorry, I could not tag your PR. " <> err
-                Right (TagName t) -> do
-                  let link = format "[{}](https://github.com/{}/{}/releases/tag/{})" (t, owner config, repository config, t)
-                  "I tagged your PR with " <> link <> ". " <>
-                    if Pr.needsDeploy approvalKind
-                    then "It is scheduled for autodeploy!"
-                    else Text.concat ["Please wait for the build of ", sha, " to pass and don't forget to deploy it!"]
-            pure pushResult
-      else
-        tryPromote prBranch newHead
-    case pushResult of
-      -- If the push worked, then this was the final stage of the pull request.
-      -- GitHub will mark the pull request as closed, and when we receive that
-      -- event, we delete the pull request from the state. Until then, reset
-      -- the integration candidate, so we proceed with the next pull request.
```

Comment:
You removed quite some comments. If they're outdated relative to the implementation, can you update them instead?

Reply:
These comments are not actually removed, but only moved, together with the code, to the promoting part of the process.